pyface/tasks/action/task_toggle_group.py

```diff
@@ -116,7 +116,7 @@ def _rebuild(self):
     # Trait initializers ---------------------------------------------------
 
     def _items_default(self):
-        self.window.on_trait_change(self._rebuild, "tasks[]")
+        self.window.observe(self._rebuild, "tasks.items")
```

Comment:
I'm not entirely sure why we choose to hook up the listener dynamically instead of simply decorating the `_rebuild` method. But that's a separate tangent - one that might be interesting to explore later on.